packages/dids/src/methods/did-dht.ts

```diff
@@ -1055,6 +1055,10 @@ export class DidDhtDocument {
 
           publicKey.alg = parsedAlg || KeyTypeToDefaultAlgorithmMap[Number(t) as DidDhtRegisteredKeyType];
 
+          if(dnsRecordId === 'k0') {
+            publicKey.kid = '0';
+          }
+
```

Comment:
Can't do this, else will break things. This is what @LiranCohen reverted. You should update the vector instead to use the thumbprint as `kid` temporarily. And optionally update #638 to mention what you changed so @LiranCohen can revert when #638 is done.

Reply:
updated